### PR TITLE
fix: update eosio::checksum256 .size() to sizeof

### PR DESCRIPTION
**Pull request by AirSwapTeam**
eosio::checksum256 value.size() is alway = 4

### CODE REVIEW

File: src/state_history_pg.hpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ inline eosio::checksum256 sql_to_checksum256(const char* ch) {
     std::vector<uint8_t> v;
     boost::algorithm::unhex(ch, ch + strlen(ch), std::back_inserter(v));
     eosio::checksum256 result;
-    if (v.size() != result.value.size())
+    if (v.size() != sizeof(result.value))
         throw std::runtime_error("hex string has incorrect length");
     memcpy(result.value.data(), v.data(), result.value.size());
```
Comment (victorj8) [inline src/state_history_pg.hpp:44]: Here it should be `sizeof(result.value)` instead of `result.value.size()` too.

Comment (AirSwapTeam) [reply src/state_history_pg.hpp:44]: that right, but it work now. so i think the sha256 not use in other wise

**Comment by AirSwapTeam** [comment]
fix #127 , when enther soft fork or restart history-tools, the process will be crashed with funcation flush_streams. cause operate table_streams

**Comment by victorj8** [comment]
The first 2 fixes were merged on:

https://github.com/EOSIO/history-tools/pull/132
https://github.com/EOSIO/history-tools/pull/133

@AirSwapTeam can you please create a new PR for the last commit you added and some explanation about what is the purpose of the change, thanks for the fixes!

**Pull request closed without merging**